src/common/netif.cpp

```diff
@@ -82,42 +94,77 @@ std::optional<CNetAddr> QueryDefaultGatewayImpl(sa_family_t family)
         return std::nullopt;
     }
 
-    // Receive response.
-    char response[4096];
-    int64_t recv_result;
-    do {
-        recv_result = sock->Recv(response, sizeof(response), 0);
-    } while (recv_result < 0 && (errno == EINTR || errno == EAGAIN));
-    if (recv_result < 0) {
-        LogPrintLevel(BCLog::NET, BCLog::Level::Error, "recv() from netlink socket: %s\n", NetworkErrorString(errno));
-        return std::nullopt;
-    }
+    // Receive response(s)
+    std::vector<char> buffer;
+    buffer.reserve(4096); // Usually large-enough for a single-part message
+    size_t total_bytes_read = 0;
+    bool done = false;
+
+    while (!done) {
+        char temp[4096];
+        int64_t recv_result;
+        do {
+            recv_result = sock->Recv(temp, sizeof(temp), 0);
+        } while (recv_result < 0 && (errno == EINTR || errno == EAGAIN));
```

Comment:
Since you're touching this line... According to the internet, we should also check `EWOULDBLOCK` even though it's usually the same as `EAGAIN`, and it's likely not relevant for any system we support.
https://stackoverflow.com/a/49421517